lib/y2_rest_api.pm

```diff
@@ -0,0 +1,42 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+package y2_rest_api;
+
+use base "y2_installbase";
+use strict;
+use warnings;
+
+use testapi;
+use YuiRestClient;
+use Carp::Always;
+use Mojo::JSON 'to_json';
+
+=head1 y2_rest_api
+
+C<y2_rest_api> - Base class for Yast installer related functionality using REST API
+
+=cut
+
+sub post_fail_hook {
+    my $self = shift;
+    save_screenshot;
+    if (get_var('YUI_REST_API')) {
+        my $json = to_json(YuiRestClient::connect_to_running_app());
+        save_tmp_file('widgets.json', $json);
+        select_console('root-console');
```

Comment:
I wonder if we should use `log-console`, also keep in mind that we can do this request from the worker directly and print things to the logs. We can use same mechanism, as we do for autoyast profiles, see`sub upload_profile`.

Reply:
good idea, I just understood how this ulogs magic folder works, makes sense.